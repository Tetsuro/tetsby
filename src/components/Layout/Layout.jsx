import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { StaticQuery, graphql } from 'gatsby';

import Header from '../Header';

import '../../stylesheets/base.scss';
import styles from './Layout.module.scss';

class Layout extends Component {
  render() {
    const { children, title, description } = this.props;

    return (
      <div className={styles.Container}>
        <Header title={title} description={description} />
        <main>{children}</main>
      </div>
    );
  }
}

export default ({ children }) => (
  <StaticQuery
    query={graphql`
      query SiteTitleQuery {
        site {
          siteMetadata {
            title
            description
          }
        }
      }
    `}
    render={data => (
      <Layout
        title={data.site.siteMetadata.title}
        description={data.site.siteMetadata.description}
      >
        {children}
      </Layout>
    )}
  />
);
