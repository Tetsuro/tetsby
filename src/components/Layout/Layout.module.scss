.Container {
  max-width: 620px;
  margin: 0 auto;
}
