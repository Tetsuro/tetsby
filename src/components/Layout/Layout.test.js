import React from 'react';
import { render } from '@testing-library/react';

import { PureLayout as Layout } from './Layout';

const mockData = {
  site: {
    siteMetadata: {
      twitter: 'mock twitter',
      github: 'mock github',
      title: 'mock title',
      description: 'mock description',
    },
  },
};

jest.mock('../MainMenu', () => 'MockMainMenu');

test('Renders `description`', () => {
  const { container, getByText } = render(<Layout data={mockData} />);
  expect(container.querySelector('hello')).toBeInTheDocument();

  // expect(getByText('mock description')).toBeInTheDocument();
});
