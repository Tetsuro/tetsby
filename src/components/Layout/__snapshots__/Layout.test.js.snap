// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders \`<Header />\` 1`] = `
<headerf
  data={
    Object {
      "site": Object {
        "siteMetadata": Object {
          "twitter": "mock twitter",
        },
      },
    }
  }
/>
`;
